### PR TITLE
Migrate @storybook/angular to typescript

### PR DESCRIPTION
**Pull request by kroeder**
Issue: https://github.com/storybooks/storybook/issues/5030

### CODE REVIEW

File: app/angular/src/server/__tests__/angular-cli_config.test.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { getAngularCliWebpackConfigOptions } from './angular-cli_config';
-import angularJson from '../../../../examples/angular-cli/angular.json';
+import { getAngularCliWebpackConfigOptions } from '../angular-cli_config';
```
Comment (igor-dv) [inline app/angular/src/server/__tests__/angular-cli_config.test.ts]: just checking what is happening around, do we have some new convention to put tests in the `__tests` dir ?

Comment (kroeder) [reply app/angular/src/server/__tests__/angular-cli_config.test.ts]: We don't. I just wanted to bring some order in the server directory. 

image

Is it worth to caring about this? I don't know. I can leave it as is 🙂

File: app/angular/package.json [hunk 2]
```diff
@@ -46,9 +49,12 @@
     "@angular/forms": ">=6.0.0",
     "@angular/platform-browser": ">=6.0.0",
     "@angular/platform-browser-dynamic": ">=6.0.0",
+    "@angular-devkit/build-angular": ">=6.0.0",
     "autoprefixer": "^8.1.0",
     "babel-loader": "^7.0.0 || ^8.0.0",
     "rxjs": "^6.0.0",
+    "typescript": "^3.4.4",
```
Comment (ndelangen) [inline app/angular/package.json]: Do we REQUIRE the .4 patch version? if not, try to keep dependencies to `.0` patch version for maximum compatibility.

Our lockfile will make sure we'll actually use the latest patch.

If we keep these version rounded down to the lowest patch, it's less likely users will end up with multiple versions installed.

**Review by ndelangen — APPROVED** [review]

File: app/angular/package.json (hunk 2 above)
Comment (kroeder) [reply app/angular/package.json]: Done

File: app/angular/src/server/__tests__/angular-cli_config.test.ts (hunk 1 above)
Comment (ndelangen) [reply app/angular/src/server/__tests__/angular-cli_config.test.ts]: I like colocation of tests, until the folder gets like 6+ files, then having a `tests` folder is kinda nice. I don't care enough about it either way really. 👍

**Pull request merged**